idea/test-balance/src/gear/index.ts

```diff
@@ -1,85 +1,57 @@
-import { initLogger, RabbitMQueues } from '@gear-js/common';
+import { initLogger } from '@gear-js/common';
 
 import config from '../config/configuration';
 import { ResponseTransferBalance } from './types';
 import { transferService } from '../services/transfer.service';
 import { createAccount } from './utils';
-import { changeStatus } from '../routes/healthcheck.router';
-import { producer } from '../rabbitmq/producer';
+import { connect, api, getGenesisHash } from './connection';
 
-let api: GearApi;
-let accountGR: KeyringPair;
-let rootAccountGR: KeyringPair;
-let accountBalanceGR: BN;
-let balanceToTransferGR: BN;
+let tbAccount: KeyringPair;
+let prefundedAcc: KeyringPair;
+let tbAccBalance: BN;
+let balanceToTransfer: BN;
 
 const logger = initLogger('TEST_BALANCE_GEAR');
 
-async function connect() {
-  api = await GearApi.create({ providerAddress: config.gear.providerAddress });
-  logger.info(`Connected to ${await api.chain()} with genesis ${getGenesisHash()}`);
-  changeStatus('ws');
+async function init() {
+  tbAccount = await createAccount(config.gear.accountSeed);
+  prefundedAcc = await createAccount(config.gear.rootAccountSeed);
+  tbAccBalance = new BN(config.gear.accountBalance);
+  balanceToTransfer = new BN(config.gear.balanceToTransfer);
 
-  if (await isSmallAccountBalance()) {
-    await transferBalance(accountGR.address, rootAccountGR, accountBalanceGR);
-  }
-}
-
-async function init(connectionEstablishedCb: () => void) {
-  accountGR = await createAccount(config.gear.accountSeed);
-  rootAccountGR = await createAccount(config.gear.rootAccountSeed);
-  accountBalanceGR = new BN(config.gear.accountBalance);
-  balanceToTransferGR = new BN(config.gear.balanceToTransfer);
-  let onInit = true;
-
-  while (true) {
-    try {
-      await connect();
-    } catch (error) {
-      console.log('Reconnecting...');
-      continue;
-    }
+  await connect();
 
-    if (onInit) {
-      connectionEstablishedCb();
-      onInit = false;
-    } else {
-      await producer.sendGenesis(RabbitMQueues.GENESISES, getGenesisHash());
-    }
-
-    await new Promise((resolve) =>
-      api.on('error', async (error) => {
-        console.log(`${new Date()} | Gear node connection error`, error);
-        await producer.sendDeleteGenesis(RabbitMQueues.GENESISES, getGenesisHash());
-        changeStatus('ws');
-        resolve(error);
-      }),
-    );
-    console.log('Reconnecting...');
+  if (await isSmallAccountBalance()) {
+    await transferBalance(tbAccount.address, prefundedAcc, tbAccBalance);
   }
 }
 
 async function transferBalance(
   to: string,
-  from: KeyringPair = accountGR,
-  balance: BN = balanceToTransferGR,
+  from: KeyringPair = tbAccount,
+  balance: BN = balanceToTransfer,
 ): Promise<ResponseTransferBalance> {
+  logger.info(`Transfer value ${balance.toNumber()} from ${from.address} to ${to}`);
   try {
-    await transfer(from, to, balance);
+    await transfer(to, from, balance);
```

Comment:
@osipov-mit What do you think before calling function `transfer` inside `transferBalance` to check 
with flag function `isSmallAccountBalance()` balance of the root account to avoid error that we get on our `idea`.

error - `[TEST_BALANCE_GEAR] [{"module":{"index":5,"error":"0x02000000"}},{"weight":{"refTime":266899000,"proofSize":0},"class":"Normal","paysFee":"Yes"}]`

Also, we can create `JSON_RPC` error for that case.

Reply:
Not a good idea actually. It will increase the time of request which has already increased in some cases due to sequential order of the requests. I have some thoughts on how to do it better but not here.